components/automate-cli/cmd/chef-automate/automateDeploymentFactory.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+type deployDirector struct {
```

Comment:
Builder pattern is used when Factory Pattern becomes too complex. or Object need to lot of steps to create. I feel Simple Factory pattern should be fine for us here.

Reply:
Agreed. This file is unnecessary and deployer is enough.